### DISCUSSION TITLE
createContext only works in Client Components.

### DISCUSSION THREAD

**Discussion by Kaseax** [heroui-inc/heroui#1784] [Help]
Just finished setting up NextUI into my NextJS project. but now I can't use the NextUI Components on Serverside. I'm getting
"Error: createContext only works in Client Components. Add the "use client" directive at the top of the file to use it. Read more: https://nextjs.org/docs/messages/context-in-server-component" on every page, I can't just add "use client" to every page.

**Comment by trungphanthanh** [comment]
create a lib folder at the same level as app in which you add a ```next-ui.ts``` file, like so:
```
"use client";
export * from "@nextui-org/react";
```

**Comment by Skarvion** [comment]
While trungphanhthanh's solution works, it's not how it's described in the official documentation.

According to https://nextui.org/docs/guide/upgrade-to-v2, you have to tweak the import path

> Important 🚨: Note that you need to import the component from the individual package, not the from @nextui-org/react.

For example:

`import { Input } from "@nextui-org/input"`

This way, you keep the code more streamlined without any intermediary file. Granted, it's a pain when you add a lot of component types as each takes 1 line.

**Reply by pppdns** [reply]
this worked for me! Thanks!

I also needed to put my `<NextUIProvider>` in a separate file, instead of using it directly in the layout, so don't skip this step in the docs:
https://nextui.org/docs/guide/upgrade-to-v2#setup-provider
